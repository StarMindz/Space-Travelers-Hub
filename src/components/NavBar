import { Link } from 'react-router-dom';
import React from 'react';

const NavBar = () => {
  const nav = (
    <div className="navBar">
      <Link className="home" to="/">Books</Link>
      <Link className="categories" to="/categories">Categories</Link>
    </div>
  );
  return nav;
};

export default NavBar;