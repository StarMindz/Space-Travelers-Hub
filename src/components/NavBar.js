import { NavLink } from 'react-router-dom';
import React from 'react';

const NavBar = () => {
  const nav = (
    <div className="navBar">
      <NavLink className="rockets" to="/">ROCKETS</NavLink>
      <NavLink className="missions" to="/missions">MISSiONS</NavLink>
      <NavLink className="profile" to="/profile">PROFILE</NavLink>
    </div>
  );
  return nav;
};

export default NavBar;
