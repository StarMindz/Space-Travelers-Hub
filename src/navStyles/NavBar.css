.navBar {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid rgb(201, 205, 211);
}

.navBarHead {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.navBarMenu {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 300px;
  margin-right: 50px;
}

.navImage {
  width: 60px;
  margin: 20px;
}

.rocketsLink,
.missionsLink,
.profileLink {
  text-decoration: none;
  font-size: 20px;
}

.active {
  color: rgb(35, 34, 44);
}