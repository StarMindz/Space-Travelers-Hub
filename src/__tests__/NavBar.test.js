import React from 'react';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../components/NavBar';

describe('Navbar component', () => {
  test('Show Correct render of Navbar', () => {
    const navbar = render(
      <Router>
        <NavBar />
      </Router>,
    );
    expect(navbar).toMatchSnapshot();
  });

  test('Check if \'Space Travelers Hub\' is in the rendered document of Navbar', () => {
    render(
      <Router>
        <NavBar />
      </Router>,
    );
    expect(screen.getByText('Space Travelers Hub')).toBeInTheDocument();
  });
});
