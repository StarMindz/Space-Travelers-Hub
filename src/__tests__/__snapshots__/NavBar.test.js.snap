// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Show Correct render of Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navBar"
      >
        <div
          class="navBarHead"
        >
          <img
            alt="Logo"
            class="navImage"
            src="planet.png"
          />
          <h1>
            Space Travelers Hub
          </h1>
        </div>
        <div
          class="navBarMenu"
        >
          <a
            activeclassname="active"
            aria-current="page"
            class="rocketsLink active"
            href="/"
          >
            ROCKETS
          </a>
          <a
            activeclassname="active"
            class="missionsLink"
            href="/missions"
          >
            MISSIONS
          </a>
          <a
            activeclassname="active"
            class="profileLink"
            href="/profile"
          >
            PROFILE
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navBar"
    >
      <div
        class="navBarHead"
      >
        <img
          alt="Logo"
          class="navImage"
          src="planet.png"
        />
        <h1>
          Space Travelers Hub
        </h1>
      </div>
      <div
        class="navBarMenu"
      >
        <a
          activeclassname="active"
          aria-current="page"
          class="rocketsLink active"
          href="/"
        >
          ROCKETS
        </a>
        <a
          activeclassname="active"
          class="missionsLink"
          href="/missions"
        >
          MISSIONS
        </a>
        <a
          activeclassname="active"
          class="profileLink"
          href="/profile"
        >
          PROFILE
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
